bravado/mapping/operation.py

```diff
@@ -54,49 +93,84 @@ def operation_id(self):
+            'params': {},
+            'headers': request_options.get('headers', {}),
+        }
+        self.construct_params(request, kwargs)
         return request
 
+    def construct_params(self, request, op_kwargs):
+        """
+        Given the parameters passed to the operation invocation, validates and
+        marshals the parmameters into the request dict.
+
+        :type request: dict
+        :param op_kwargs: the kwargs passed to the operation invocation
+        :raises: TypeError on extra parameters or when a required parameter
+            is not supplied.
+        """
+        current_params = self.params.copy()
+        for param_name, param_value in op_kwargs.iteritems():
+            param = current_params.pop(param_name, None)
+            if param is None:
+                raise TypeError("{0} does not have parameter {1}".format(
+                    self.operation_id, param_name))
+            marshal_param(param, param_value, request)
+
+        # Check required params and non-required params with a 'default' value
+        for remaining_param in current_params.itervalues():
+            if remaining_param.required:
+                raise TypeError('{0} is a required parameter'.format(remaining_param.name))
+            if not remaining_param.required and remaining_param.has_default():
+                marshal_param(remaining_param, None, request)
+
     def __call__(self, **kwargs):
-        log.debug(u"%s?%r" % (self.operation_id, urllib_utf8.urlencode(kwargs)))
-        request = self._construct_request(**kwargs)
+        # TODO: rewrite/simplify
+        log.debug(u"%s(%s)" % (self.operation_id, kwargs))
+        request = self.construct_request(**kwargs)
 
         def response_future(response, **kwargs):
             # Assume status is OK, an exception would have been raised already
             if not response.text:
                 return None
 
-            print response.json()
+            status_code = str(response.status_code)
+            # Handle which repsonse to activate given status_code
+            default_response_spec = self.operation_spec['responses'].get('default', None)
+            response_spec = self.operation_spec['responses'].get(status_code, default_response_spec)
+            if response_spec is None:
+                # reponse code doesn't match and no default provided
+                if status_code == '200':
+                    # it was obviously successful
+                    log.warn("Op {0} was successful by didn't match any responses".format(self.operation_id))
+                else:
+                    raise SwaggerError("Response doesn't match any expected responses: {0}".format(response))
+
+            response_dict = response.json()
+
+            if response_spec and 'schema' in response_spec:
+                swagger_type_ = swagger_type.get_swagger_type(response_spec['schema'])
+            else:
+                swagger_type_ = None
```

Comment:
line 151 to here should probably be extracted as a new function `get_swagger_response_type()` or something like that since this function is starting to get pretty big.